generate_markdown_from_doxygen_xml.py

```diff
@@ -0,0 +1,1008 @@
+"""
+This script converts Doxygen XML output into gitbook-compatible markdown. There are no arguments.
+
+- The directories that are parsed for source/outputs is DOXYGEN_XML_DIR (set DOXYGEN_ROOT_DIR for path from this file execution)
+- The list of files to parse are in the set: FILES_TO_PROCESS
+
+How it works:
+- There are Python objects for each C++ type - e.g. cppClass, cppTypdef etc. (cppClass covers both class and struct)
+- Each object has a method import_from_doxygen_class_file() which takes a Tag (or filename for CPP) and parses in data. 
+  - The cpp class takes a file, and creates the sub objects it owns (e.g. methods), calling their import_from_doxygen_class_file() method with a tag.
+- Each object (mostly) has a markdown_overview() and markdown() method that renders the markdown from the respective objects
+- markdown_any_tag() - 
+  - Some objects/values will be empty and others will store tagged information that is only partially parsed from XML (in theory this allows us 
+  to change the output rendering, because we store it in generic format. The markdown_any_tag() method can be used to get 
+  markdown text or an empty string from any stored attribute. 
+  - This can be safely called on any "data" type attributed
+  - This will report an error on console if an unsupported tag is detected. The tag will render in output anyway, but may contain odd tags.
+- cleanup_markdown_string
+  - Just cleans up a little of the markdown.
+
```

Comment:
Did you just write this script? If yes, I'll review it in more depth.

Reply:
Yes :-) The output needs review more than the architecture (not saying it is great, but it should be OK for the stage we're at now).